python/Ganga/Core/InternalServices/ShutdownManager.py

```diff
@@ -74,6 +74,8 @@ def _ganga_run_exitfuncs():
         from Ganga.Core.MonitoringComponent.Local_GangaMC_Service import getStackTrace
         getStackTrace()
         monitoring_component.disableMonitoring()
+        monitoring_component.stop()
+        monitoring_component.join()
```

Comment:
I had actually thought to put this lower down, just before the registries are shutdown (to be safer) but if it works here then that's OK as well :smile: